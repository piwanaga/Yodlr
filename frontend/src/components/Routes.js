/**Routes component */

import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Home from './Home';
import RegisterForm from './RegisterForm';
import Admin from './Admin';

const Routes = () => {
    return (
        <Switch>
            <Route path='/register'>
                <RegisterForm />
            </Route>
            <Route path='/admin'>
                <Admin />
            </Route>
            <Route path ='/'>
                <Home />
            </Route>
        </Switch>
    )
};

export default Routes