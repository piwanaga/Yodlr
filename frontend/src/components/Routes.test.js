import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Routes from './Routes';

describe('Routes', () => {
  let Wrapper

  beforeEach(() => {
      Wrapper = ({ children }) => (
          <BrowserRouter>
              {children}
          </BrowserRouter>
      );
  });

  it("passes snapshot test", () => {
    const { asFragment } = render(<Routes />, {wrapper: Wrapper});
    expect(asFragment()).toMatchSnapshot()
  });
});