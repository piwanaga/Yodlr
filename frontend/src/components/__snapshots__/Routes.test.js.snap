// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm passes snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <a
        class="MuiTypography-root MuiTypography-body1"
        href="/"
      >
        Back to Home
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <h5
        class="MuiTypography-root makeStyles-header-1 MuiTypography-h5 MuiTypography-alignCenter"
      >
        Register
      </h5>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <form
        class="makeStyles-form-2"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="email"
                id="email-label"
              >
                Email
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="email"
                  name="email"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-5"
                  >
                    <span>
                      Email
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="firstName"
                id="firstName-label"
              >
                First Name
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="firstName"
                  name="firstName"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-5"
                  >
                    <span>
                      First Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="lastName"
                id="lastName-label"
              >
                Last Name
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="lastName"
                  name="lastName"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-5"
                  >
                    <span>
                      Last Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Register!
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Routes passes snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
  >
    <div
      class="MuiGrid-root makeStyles-header-1 MuiGrid-item"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2"
      >
        Yodlr Homepage
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-item MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <a
          class="MuiTypography-root MuiTypography-body1"
          href="/register"
        >
          Register
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <a
          class="MuiTypography-root MuiTypography-body1"
          href="/admin"
        >
          Admin
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
